sotodlib/toast/ops/sim_hwpss.py

```diff
@@ -190,9 +265,28 @@ def _exec(self, data, detectors=None, **kwargs):
 
                 iquss -= np.median(iquss)
 
+                if self.hwpss_random_drift:
+                    # Apply detector couplings to HWPSS random drift common mode
+                    key1 = obs.telescope.uid
+                    key2 = obs.session.uid
+                    counter1 = self.realization
+                    counter2 = focalplane[det]["uid"]
+                    gaussian = toast.rng.random(
+                        1,
+                        sampler="gaussian",
+                        key=(key1, key2),
+                        counter=(counter1, counter2),
+                    )[0]
+                    coupling = (
+                        self.hwpss_drift_coupling_center
+                        + gaussian * self.hwpss_drift_coupling_width
+                    )
+                else:
+                    coupling = 1.0
+                
                 # Co-add with the cached signal
 
-                signal += det_scale * iquss
+                signal += det_scale * iquss * drift * coupling
```

Comment:
If the coupling truly applies to the drifting portion of the IQUSS signal, then it cannot be applied like this. Rather, you would need
```
signal += det_scale * iquss * (1 + (drift - 1) * coupling)
```
We could of course redefine `drift` to have zero mean, in which case
```
signal += det_scale * iquss * (1 + drift * coupling)
```
would do.